browser/brave_rewards/android/brave_rewards_native_worker.cc

```diff
@@ -109,6 +112,30 @@ bool BraveRewardsNativeWorker::IsRewardsEnabled(JNIEnv* env) {
       ->GetBoolean(brave_rewards::prefs::kEnabled);
 }
 
+bool BraveRewardsNativeWorker::ShouldShowSelfCustodyInvite(JNIEnv* env) {
+  if (base::FeatureList::IsEnabled(
+          brave_rewards::features::kAllowSelfCustodyProvidersFeature)) {
```

Comment:
Instead of digging into this pref, you can call `RewardsServiceImpl::GetExternalWalletProviders`. If solana is allowed (taking into account the "self_custody_available" pref), then it will be in the list.

You also don't need the feature test, since `GetExternalWalletProviders` will take that into account as well.